### ISSUE TITLE
vcpkg remove is case sensitive, but install is not

### ISSUE THREAD

**Issue by chwarr** [Microsoft/vcpkg#103]
I can use any casing I want to install a package, but it appears that I must use the exact casing of the Source: line in the CONTROL file when removing.

```
> .\vcpkg install zLIb
-- CURRENT_INSTALLED_DIR=S:/src/vcpkg/installed/x86-windows
-- DOWNLOADS=S:/src/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=S:/src/vcpkg/packages/zLIb_x86-windows
-- CURRENT_BUILDTREES_DIR=S:/src/vcpkg/buildtrees/zLIb
-- CURRENT_PORT_DIR=S:/src/vcpkg/ports/zLIb/.
-- Using cached S:/src/vcpkg/downloads/zlib128.zip
-- Testing integrity of cached file...
-- Testing integrity of cached file... OK
-- Extracting done
-- Configuring x86-windows-rel
-- Configuring x86-windows-rel done
-- Configuring x86-windows-dbg
-- Configuring x86-windows-dbg done
-- Build x86-windows-rel
-- Build x86-windows-rel done
-- Build x86-windows-dbg
-- Build x86-windows-dbg done
-- Package x86-windows-rel
-- Package x86-windows-rel done
-- Package x86-windows-dbg
-- Package x86-windows-dbg done
-- Installing: S:/src/vcpkg/packages/zLIb_x86-windows/share/zlib/copyright
-- Warning: Could not find a matching pdb file for:
    S:/src/vcpkg/packages/zLIb_x86-windows/bin/zlib.dll

-- Performing post-build validation
-- Performing post-build validation done
Package zLIb:x86-windows is installed
> .\vcpkg remove Zlib
Package Zlib:x86-windows is not installed
> .\vcpkg remove zLIb
Package zLIb:x86-windows is not installed
> .\vcpkg remove zlib
Package zlib:x86-windows was successfully removed
```

Version: Vcpkg package management program version 0.0.30-6273b1d1702f4ec1a356c99dac9ae3862f5b73e7

**Comment by ras0219-msft** [comment]
We should require package names (in files) to always be lowercase ascii, then we can easily canonicalize all package name inputs on the command line.

**Comment by alexkaratarakis** [comment]
Thanks for reporting!
This has been fixed after 1f9b97d683d65e848c70121a46605a331fe75648.
The behavior is as follows:
User-input is lowercased, so casing on the commands does not matter. Therefore all casing variations of: `vcpkg <command> zLiB` are accepted by `vcpkg` and are treated as `vcpkg <command> zlib`

However, port files are required to use lowercase chars, e.g. zlib would be required to be named as such and zLib would be an error.
`vcpkg` checks that the port names only include lower-case, alphanumeric chars or dashes.
